text/0000-flatten-feature.md

```diff
@@ -0,0 +1,222 @@
+
+The new flag to be included is:
+
+- `--flatten=<buildpacks>` will flatten the Buildpacks specified after the `flatten` flag into a single layer. Can be used more than once, with each use resulting in a single layer.
+
+We also need to define how a Platform implementor needs to consume a flattened Builder.
+
+- When a Platform consumes a Builder, they will need to inspect each Buildpack layer blob and determine if the blob contains more than one Buildpack, in such as case, they will need to process those Buildpacks correctly.
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Let's say we have a Composite Buildpack (CB1) with the following dependency tree:
+```mermaid
+flowchart TD
+    A[CB1]
+    A --> B[G1]
+    A --> C[G2]
+    B --> BPA[BP1]
+    B --> BPFOO[BP2]
+    B --> BPC[BP4]
+    C --> BPD[BP1]
+    C --> BPBAR[BP3]
+    C --> BPE[BP4]
+```
+
+Until now, when a Buildpack like this is being shipped into a Builder every individual Buildpack is being saved in one layer, as a result we will have:
+
+$$
+layer_1 = [CB_1] \\
+layer_2 = [G_1] \\
+layer_3 = [BP_1] \\
+layer_4 = [BP_2] \\
+layer_5 = [BP_4] \\
+layer_6 = [G_2] \\
+layer_7 = [BP_3] \\
+total = \text{7 layers}
+$$
+
+Noticed that duplicated Buildpacks are cleaned up.
+
+We can use the new `flatten` flag to reduce the number of Builder layers used by the buildpacks in different ways.
+
+* `--flatten=<buildpacks>` i.e. `--flatten=<BP1,BP2> --flatten=<BP3,BP4>`:
+  Will group the given Buildpacks into one layer and keep the other ones as single layers Buildpacks, the result will be:
+
+```mermaid
+classDiagram
+    class Layer1 {
+      CB1
+    }
+    class Layer2 {
+      G1 
+    }
+    class Layer3 {
+      BP1
+      BP2
+    }
+    class Layer4 {
+      G2
+    }
+    class Layer5 {
+      BP3
+      BP4
+    }
+```
+
+
+$$
+total = \text{5 layers}
+$$
+
+---
+
+
+# Migration
+[migration]: #migration
+
+
+The current [distribution spec](https://github.com/buildpacks/spec/blob/main/distribution.md#buildpackage) defines:
+
+```
+Each buildpack layer blob MUST contain a single buildpack at the following file path:
+
+/cnb/buildpacks/<buildpack ID>/<buildpack version>/
+```
+
+A Builder flattened with this new feature would not be consumable by older platform implementations because they are not expecting to find more than one buildpack on a blob layer.
+
+
+<!--
+This section should document breaks to public API and breaks in compatibility due to this RFC's proposed changes. In addition, it should document the proposed steps that one would need to take to work through these changes. Care should be give to include all applicable personas, such as platform developers, buildpack developers, buildpack users and consumers of buildpack images.
+-->
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+It could create artifacts that are not consumable by older platforms.
```

Comment:
This use case, only happens for platforms that create ephemeral builders (like `pack` and kpack) right? I was looking at how we consume builders at Heroku and I think this change would "just work".